src/app/static/src/app/components/outputTable/TableDisplay.vue

```diff
@@ -1,14 +1,146 @@
 <template>
-    <div>Table</div>
+    <ag-grid-vue style="height: 700px"
+                 class="ag-theme-alpine"
+                 :defaultColDef="defaultColDef"
+                 :columnDefs="columnDefs"
+                 :rowData="data"
+                 @grid-ready="onGridReady">
+    </ag-grid-vue>
 </template>
 
 <script lang="ts">
-import { defineComponent } from "vue";
+import { ref, defineComponent, computed, onUpdated } from "vue";
+import { AgGridVue } from "ag-grid-vue3";
+import { AgGridEvent } from "ag-grid-community";
+import "ag-grid-community/styles//ag-grid.css";
+import "ag-grid-community/styles//ag-theme-alpine.css";
+import { useStore } from "vuex";
+import { RootState } from "../../root";
+import { formatOutput } from "../plots/utils";
+
+const defaultColDef = {
+    // Set the default filter type
+    filter: 'agNumberColumnFilter',
+    // Floating filter adds the dedicated row for filtering at the bottom
+    floatingFilter: true,
+    // suppressMenu hides the filter menu which showed on the column title
+    // this just avoids duplication of filtering UI as we have floating turned on
+    // there are some cases where other thing show in the menu but not for our example
+    suppressMenu: true,
+    // Show an icon when the column is not sorted
+    unSortIcon: true,
+    // Make column sortable
+    sortable: true,
+    // Stop the columns from being draggable to rearrange order or remove them
+    suppressMovable: true,
+    // our auto resize will apply to columns not shown on the screen
+    // e.g. if they are off to the side, we wont get auto resize
+    // because ag grid does this automatically
+    suppressColumnVirtualisation: true
+};
 
 export default defineComponent({
     props: {
         data: Object,
         headerName: String
+    },
+    components: {
+        AgGridVue
+    },
+    setup(props) {
+        const gridApi = ref<AgGridEvent | null>(null);
+
+        const ensureColumnsWideEnough = (event: AgGridEvent) => {
+            event.columnApi.autoSizeAllColumns();
+            const columns = event.columnApi.getAllGridColumns();
+            const columnLimits = columns.map(col => {
+                return { key: col.getColId(), minWidth: col.getActualWidth() };
+            });
+            event.api.sizeColumnsToFit({ columnLimits })
+        };
```

Comment:
yep can leave comments, but this is quite fiddly actually, so it will either spread columns to fill the whole space exactly which means `...` everywhere the values dont fit or you tell it to auto size the columns and it will resize them perfectly but wont fill the space so you get tables for prevalence for example which are much smaller than the available space, ag grid has nothing in between these :cry: 

but yes, will leave a comment, basically i resize, record the min widths required to fit all the data in that columns and i tell ag grid to fill the space with at least that min width in mind